### PR TITLE
scripts: dts: Add generation of *_PARENT_LABEL macros

### LINKED ISSUE #13313: Generate hierarchy while parsing DTS
We need an abstracted device hierarchy in the Zephyr in order to track device dependencies in device power management subsystem. We have to be able to do the following:
```
const char *parent_name = device_get_parent_name(device);
strucy device *parent = device_get_binding(parent_name);
```

Part of the job is alredy done: https://github.com/zephyrproject-rtos/zephyr/pull/13129.
Now we have to extract the data from DTS.

### PR DESCRIPTION
**Pull request by anangl**
This patch adds generation of `*_PARENT_LABEL` macros for nodes that have a parent with a 'label' property and also have a 'label' property themselves.
Such feature was requested by @pizi-nordic in order to create the device hierarchy needed in the power management rework he's doing.

Fixes: https://github.com/zephyrproject-rtos/zephyr/issues/13313

### CODE REVIEW

**Comment by anangl** [comment]
Rephrased the commit description.

File: scripts/dts/extract_dts_includes.py
Comment (ulfalizer) [inline scripts/dts/extract_dts_includes.py]: `get_parent_path('/')` currently returns `'/'`, so the `parent_path` check will always succeed.

Maybe the check could be `node_path != '/'` instead, if it's needed.

**Review by ulfalizer — CHANGES_REQUESTED** [review]

File: scripts/dts/extract_dts_includes.py
Comment (ulfalizer) [reply scripts/dts/extract_dts_includes.py]: Returning `None` for `/` might be best. Just ran into another case where it's useful.

Comment (ulfalizer) [reply scripts/dts/extract_dts_includes.py]: Did that in https://github.com/zephyrproject-rtos/zephyr/pull/13922. Just added

```
    if node_path == '/':
        return None
```

to the start of `get_parent_path()`.

Comment (anangl) [reply scripts/dts/extract_dts_includes.py]: > Did that in #13922.

Okay. So this PR just needs to wait until #13922 gets merged.
Actually, even with the current master everything will work. `parent_path` will always succeed, but for top-level nodes `reduced['/']` won't have the 'label' property.

Comment (anangl) [reply scripts/dts/extract_dts_includes.py]: I have just took another look. Checking `parent_path` here doesn't make much sense, as it will never be `None`. This code is executed only for nodes that have the `compatible` property, hence for path `'/'` we won't get here. I think I'll remove this check.

**Comment by anangl** [comment]
Rebased on master.

**Comment by anangl** [comment]
Removed the no longer needed `parent_path` check.

**Review by ulfalizer — APPROVED** [review]
Looks fine to me.

File: scripts/dts/extract_dts_includes.py
Comment (ulfalizer) [reply scripts/dts/extract_dts_includes.py]: Yeah, no need to avoid merge conflicts with those cleanups PRs either.

**Comment by anangl** [comment]
- rebased on master
- renamed the parameter added to `extract_parent_label`: `label_postfix` -> `def_label_postfix`

**Comment by nashif** [comment]
@anangl  are you going to update this PR? Is this still relevant?

**Comment by anangl** [comment]
> @anangl are you going to update this PR? Is this still relevant?

No, I'll close it.

**Pull request closed without merging**
